lib/client/mfa.go

```diff
@@ -128,31 +131,38 @@ func PromptMFAChallenge(
 	}
 
 	// Fire TOTP goroutine.
+	stdin := &stdinHijack{}
 	if hasTOTP {
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
 			const kind = "TOTP"
-			var msg string
 			if !quiet {
 				if hasWebauthn {
-					msg = fmt.Sprintf("Tap any %[1]ssecurity key or enter a code from a %[1]sOTP device", promptDevicePrefix, promptDevicePrefix)
+					fmt.Fprintf(os.Stderr, "Tap any %[1]ssecurity key or enter a code from a %[1]sOTP device\n", promptDevicePrefix, promptDevicePrefix)
 				} else {
-					msg = fmt.Sprintf("Enter an OTP code from a %sdevice", promptDevicePrefix)
+					fmt.Fprintf(os.Stderr, "Enter an OTP code from a %sdevice\n", promptDevicePrefix)
 				}
 			}
-			code, err := prompts.OTP(ctx, os.Stderr, prompt.Stdin(), msg)
-			if err != nil {
-				respC <- response{kind: kind, err: err}
+
+			stdin.startRead()
+			select {
+			case <-ctx.Done():
```

Comment:
If `ctx.Done()` is first we can potentially leave terminal in bad state (waiting for password even though request timed out). 
It can also leak goroutine as we don't use `ctx` in `startRead` and `prompt`